### ISSUE TITLE
Spanish tree issues with new grammar skills

### ISSUE THREAD

**Issue by immutability** [ToranSharma/Duo-Strength#111]
Using 1.3.34. My Spanish tree was recently upgraded with new grammar skills that have been put into place all over the tree. This seems to be throwing off Duo Strength - there are several issues.

1. Incorrect list of skills that need strengthening. Please see the first screenshot. The tree highlights the skill "Shopping" as the one that is on 75% but it says that the "Pres Tense 1" needs strengthening. The other 2 skills on the list are pointing to correct skills this time. I think I've seen the same issue with "cracked skills" but I'm not sure.

![Screenshot 2020-10-17 12 27 43](https://user-images.githubusercontent.com/1141653/96334927-b68ee180-1074-11eb-949d-daa0706d6492.png)

2. The "words" bubble doesn't work correctly with these new grammar skills.

![Screenshot 2020-10-17 12 28 13](https://user-images.githubusercontent.com/1141653/96334931-babaff00-1074-11eb-8d3d-9a7b67228f42.png)

Please let me know if you need more input from me or help with testing. I am not sure if these new grammar skills are being rolled out gradually or a part of an A/B test. They also seem to carry a number of new question types and overall I find them a bit annoying (too many click type questions, etc). I have already reported two issues to DL directly, perhaps they are not fully ironed out yet.

**Comment by ToranSharma** [comment]
Thanks for the report.

I'll see if I have access to these new grammar skills and see the same issues. I had been anticipating some issues since I saw them appear on the android app a week or two ago.

**Comment by immutability** [comment]
I have a small correction - it looks like it was actually the "Pres Tense 1" skill that needed strengthening. I did 4 lessons on "Shopping" (pushing it to level 5) but it remained on 75% - after doing one lesson of "Pres Tense 1" the bar under "Shopping" jumped back to 100%.

So the skill listed on the top is correct, but the actual strength bar is drawn under the wrong skill icon.

If you need any help or if I can extract the data for you somehow, let me know. Thanks for your work on this :-)

**Comment by ToranSharma** [comment]
Hi again, just been able to check on my end, and my Spanish tree has not had the update, but another account I have access to has so I can do testing with that.

I am seeing the same issues with you, the strength bars have been swapped for the skill before and the grammar skill.

The words list is also not helpful for these skills, there is likely little I can do about changing it to something more useful so I will just remove the button from such skills.

**Comment by immutability** [comment]
That's good news :-) I agree about hiding the word list.

Also, not sure if it affects DuoStrength in any way, but the grammar skills are only 2 levels - once you reach the 2nd level it turns gold. It seems DuoLingo itself has a bug here as it told me I'm "acing the skill" and offered me to test out and jump straight to level 3 right after completing level 2. Oddly enough these skills do not normally show the "Key" button to test-out manually...

**Comment by ToranSharma** [comment]
Okay, thanks for the information, seems like a few things are broken with them. Looking into these skills in the userData file, the thing causing the strength bar swapping is that the y coordinate of the skills in the data file are off by one. This is making them seem as though they are embedded into the row above and messing up the strength bars for the other skills in that row.

Should be a simple enough fix for the time being, I will just add a large number to the x coordinate, making them seem like they are at the very end of that row, and so at least correctly placed relative to them.

Here is how it looks before and after this change:
<img src="https://user-images.githubusercontent.com/4559968/96350369-7a2ba780-10ad-11eb-99a3-3f7cfd2b098b.png" alt="Grammar skills before" width="40%" /> <img alt="arrow" src="https://user-images.githubusercontent.com/4559968/96350779-48b3db80-10af-11eb-8f43-223a2c02dd26.png"> <img src="https://user-images.githubusercontent.com/4559968/96350372-7d269800-10ad-11eb-9b5e-00ba737dd9be.png" alt="Grammar skills after" width="40%" />

**Comment by ToranSharma** [comment]
I have made the above change in 25f6a20, and removed the words list in add9cf6.

Since to mention the ability to test out the grammar skills (up to a max of crown level 2), I thought it odd that it was missing the test out key button, so I have added an option to add this where it should be. It is enabled by default. Here is how it looks, should be more or less seamless.

<img alt="Grammar Skill Missing Test Out Button" src="https://user-images.githubusercontent.com/4559968/96352442-114b2c00-10bb-11eb-9545-72f29b9c44ca.png" width="40%" /><img alt="arrow" src="https://user-images.githubusercontent.com/4559968/96350779-48b3db80-10af-11eb-8f43-223a2c02dd26.png"><img alt="Grammar Skill Test Out Button" src="https://user-images.githubusercontent.com/4559968/96352413-dc3ed980-10ba-11eb-8277-26f8f467d294.png" width="47%" />

This change was made in 957d780.
